service/catalog.go

```diff
@@ -84,16 +139,28 @@ func updateCatalog(w http.ResponseWriter, r *http.Request, envId string) (int, e
 		return http.StatusBadRequest, err
 	}
 
-	if err := db.Model(&model.CatalogModel{}).Where(&model.CatalogModel{
+	if isDuplicateGlobalName(catalogModel) {
+		return http.StatusUnprocessableEntity, errors.New("Cannot add catalog with duplicate field `name`")
```

Comment:
Can you change this error message to "Catalog "foo" already exists"? I think that will be clearer and let the user know they need to change the name.